packages/mdc-tab-bar/foundation.js

```diff
@@ -0,0 +1,365 @@
+      getScrollPosition: () => {},
+      getScrollContentWidth: () => {},
+      getOffsetWidth: () => {},
+      isRTL: () => {},
+      activateTabAtIndex: () => {},
+      deactivateTabAtIndex: () => {},
+      getTabIndicatorClientRectAtIndex: () => {},
+      getTabDimensionsAtIndex: () => {},
+      getActiveTabIndex: () => {},
+      getIndexOfTab: () => {},
+      getTabListLength: () => {},
+    });
+  }
+
+  /**
+   * @param {!MDCTabBarAdapter} adapter
+   * */
+  constructor(adapter) {
+    super(Object.assign(MDCTabBarFoundation.defaultAdapter, adapter));
+  }
+
+  init() {
+    const activeIndex = this.adapter_.getActiveTabIndex();
+    this.scrollIntoView(activeIndex);
+  }
+
+  /**
+   * Activates the tab at the given index
+   * @param {number} index
+   */
+  activateTab(index) {
+    const previousActiveIndex = this.adapter_.getActiveTabIndex();
+    if (!this.indexIsInRange_(index) || index === previousActiveIndex) {
+      return;
+    }
+
+    this.adapter_.deactivateTabAtIndex(previousActiveIndex);
+    this.adapter_.activateTabAtIndex(index, this.adapter_.getTabIndicatorClientRectAtIndex(previousActiveIndex));
+    this.scrollIntoView(index);
+  }
+
+  /**
+   * Handles the keydown event
+   * @param {!Event} evt
+   */
+  handleKeyDown(evt) {
+    const keyboardNavigation = [
+      'ArrowLeft',
+      'ArrowRight',
+      'Home',
+      'End',
+    ];
+
+    // Early exit if the event key isn't one of the keyboard navigation keys
+    if (keyboardNavigation.indexOf(evt.key) === -1) {
+      return;
+    }
+
+    evt.preventDefault();
+    this.activateTabFromKeydown_(evt);
+  }
+
+  /**
+   * Handles the MDCTab:interacted event
+   * @param {!Event} evt
+   */
+  handleTabInteraction(evt) {
+    this.activateTab(this.adapter_.getIndexOfTab(evt.detail.tab));
+  }
+
+  /**
+   * Scrolls the tab at the given index into view
+   * @param {number} index The tab index to make visible
+   */
+  scrollIntoView(index) {
+    // Early exit if the index is out of range
+    if (!this.indexIsInRange_(index)) {
+      return;
+    }
+
+    // Always scroll to 0 if scrolling to the 0th index
+    if (index === 0) {
+      return this.adapter_.scrollTo(0);
+    }
+
+    // Always scroll to the max value if scrolling to the Nth index
+    // MDCTabScroller.scrollTo() will never scroll past the max possible value
+    if (index === this.adapter_.getTabListLength() - 1) {
+      return this.adapter_.scrollTo(this.adapter_.getScrollContentWidth());
+    }
+
+    if (this.isRTL_()) {
+      return this.scrollIntoViewRTL_(index);
+    }
+
+    this.scrollIntoView_(index);
+  }
+
+  /**
+   * Private method for activating a tab from the keydown
+   * @param {!Event} evt The keydown event
+   * @private
+   */
+  activateTabFromKeydown_(evt) {
+    const isRTL = this.isRTL_();
+    const maxTabIndex = this.adapter_.getTabListLength() - 1;
+    const shouldGoToEnd = evt.key === 'End' && !isRTL || evt.key === 'Home' && isRTL;
```

Comment:
I don't think we should be reversing the meaning of home and end in RTL. RTL reverses left and right, but the notion of which tab is first and which tab is last shouldn't change, right?

Reply:
Makes sense. Lynn and I had a similar conversation. I'll update this.